### PR TITLE
MyLocationLayer: add an optional label next to the symbol

### PR DESCRIPTION
**Pull request by janusw**
This PR implements the feature suggested in #1143: It adds an optional label to the MyLocationLayer, next to the actual location symbol. The label text can be chosen freely, so it can for example be used to display the coordinates of the user's location (or any other information). The label is hidden by default and can be shown or hidden as desired. Also clicks on the MyLocation symbol can be caught now, e.g. in order to toggle the label when clicked.

I extended Mapsui.Samples.Forms to display a label with coordinates when the MyLocation symbol is clicked, so anyone can test the feature in the sample apps.

Any kind of feedback welcome ...

### CODE REVIEW

**Comment by janusw** [comment]
> I saw, you made a new Feature for the direction, but normally it should be enough, to add another style. So the user location would be one Feature with three styles: one for the user location itself, one for the direction and one for an info text/callout.

I'll start with this one: Very good point! (Note that I'm still on the learning curve in Mapsui's guts, and don't know the API very well yet.)

I rebased the branch to current master and added a commit that fixes this.

**Comment by janusw** [comment]
> So, what is more common: a label with a white background following your position or a callout to see what is your current position?

Actually a callout style might indeed make more sense here. It looks nicer and offers more possibilites.

However, my first attempt to use a callout style somehow failed. I'll try again later.


> Both are styles, so why not expose a InfoStyle, which could be set by the user if the default isn't good enough?

Hm, I'm not sure if it's necessary to expose the complete style. We don't do that with the other two styles in the MyLocationLayer either. The user obtains full control then, but also more opportunity to break stuff  ;)

Let me think about this ...

**Comment by janusw** [comment]
> > So, what is more common: a label with a white background following your position or a callout to see what is your current position?
> 
> Actually a callout style might indeed make more sense here. It looks nicer and offers more possibilites.
> 
> However, my first attempt to use a callout style somehow failed. I'll try again later.

I suspect this might be related to #1496. Any hints on this are very welcome ...

**Comment by janusw** [comment]
> > So, what is more common: a label with a white background following your position or a callout to see what is your current position?
> 
> Actually a callout style might indeed make more sense here. It looks nicer and offers more possibilites.
> 
> However, my first attempt to use a callout style somehow failed. I'll try again later.

Alright, this works now. I'm switching to CalloutStyle in dad21d76d21a171c24148479fdaca5dfe6cff0d5.

> > Both are styles, so why not expose a InfoStyle, which could be set by the user if the default isn't good enough?
> 
> Hm, I'm not sure if it's necessary to expose the complete style. We don't do that with the other two styles in the MyLocationLayer either. The user obtains full control then, but also more opportunity to break stuff ;)

Actually I'd rather stay with a simple interface for now, at the expense of full flexibility (unless you have a strong preference to have a configurable style). If anyone using this needs more flexibility, we can still extend it later.

From my side this PR is ready now. I hope I have taken care of all your comments, @charlenni. Ok for master like this?

File: Mapsui.UI.Forms/Objects/MyLocationLayer.cs [hunk 1]
```diff
@@ -89,6 +91,44 @@ public bool IsMoving
         /// <value>Scale of symbol</value>
         public double Scale { get; set; } = 1.0;
 
+        /// <summary>
+        /// The text that is displayed in the MyLocation callout.
+        /// (can contains line breaks).
```
Comment (charlenni) [inline Mapsui.UI.Forms/Objects/MyLocationLayer.cs:96]: Seems to be a typo: contain instead of contains

**Comment by charlenni** [comment]
For me it looks good. And yes, I think we could assume, that developers with special needs create their own special MapControl, which could handle all, what they need.

**Comment by janusw** [comment]
> For me it looks good.

Cool 😃, is that an approval?

(Btw, I'm not sure where the GitHub Actions failures come from, but they occur also on master ...)

**Review by charlenni — APPROVED** [review]
Yes, that is an approval. My first one, if it works.

**Comment by charlenni** [comment]
Yes, the failing action is because of UWP. Got the same error with my PR.

**Comment by janusw** [comment]
> Yes, that is an approval. My first one, if it works.

It does 🙂, thank you. I just merged it.

**Pull request merged**